### PR TITLE
Extension RFC and Directory Changes.

### PR DESCRIPTION
**Pull request by sr229**
Initial version of the Extension RFC. This also structures our directory sorted by date of publication.

### CODE REVIEW

File: client/2022-03-07/rfc-extensions-io.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+export function activate() {
+    const fileBuf = fs.readFile('./data.txt');
+    if (typeof fileBuf === Blob)
+      vignette.notification.send('info', 'Successfully read data.txt');
+    else 
+      vignette.notification.send('error', `Expected Blob, got ${typeof fileBuf} instead.`)
+}
+```
+
+The filesystem API only allows basic file operations, `readFile`, `unlink`, and `writeFile` and only takes a `Buffer` or a `Blob` as a input.
```
Comment (LeNitrous) [inline client/2022-03-07/rfc-extensions-io.md:39]: Upon checking, `Blob` is a part of the File API and is not part of the vanilla V8.

File: client/2022-03-07/rfc-extensions-io.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+* Do NOT grant host level filesystem access to the extension.
+* Present a limited API that covers common use-cases.
+* Do NOT allow the extension to manage OS-specific file-related attributes.
+
+### API
+
+Vignette's FS system will be a reduced version of the FS API of Node.js. A basic file operation may look something like this.
+
+```typescript
+import { vignette, fs } from 'vignette';
```
Comment (LeNitrous) [inline client/2022-03-07/rfc-extensions-io.md:28]: We can split off `fs` to its own module:
```js
import { fs } from "fs";
```

File: client/2022-03-07/rfc-extensions-io.md (hunk 1 above)
Comment (sr229) [reply client/2022-03-07/rfc-extensions-io.md]: Would you want to implement `Buffer` instead? We can represent this internally as a `IOStream`.

**Comment by LeNitrous** [comment]
I find it weirdly inconsistent that `fetch` is in the global scope while `fs` is in a module. Sure we're trying to emulate nodejs and browser here but the inconsistency is just uncanny.

**Pull request merged**